components/org.wso2.carbon.identity.query.saml/src/main/java/org/wso2/carbon/identity/query/saml/validation/AbstractSAMLQueryValidator.java

```diff
@@ -0,0 +1,221 @@
+
+        boolean isIssuerValidated;
+        boolean isSignatureValidated;
+        boolean isValidSAMLVersion;
+
+        try {
+            //validate SAML Request vertion
+            isValidSAMLVersion = this.validateSAMLVersion(request);
+
+            if (isValidSAMLVersion) {
+                //validate Issuer of Request
+                isIssuerValidated = this.validateIssuer(request);
+            } else {
+                //invalid SAML version
+                invalidItems.add(new InvalidItemDTO(SAMLQueryRequestConstants.ValidationType.VAL_VERSION,
+                        SAMLQueryRequestConstants.ValidationMessage.VAL_VERSION_ERROR));
+                log.error(SAMLQueryRequestConstants.ValidationMessage.VAL_VERSION_ERROR);
+                return false;
+
+            }
+
+            if (isIssuerValidated) {
+                //validate Signature of Request
+                isSignatureValidated = this.validateSignature(request);
+            } else {
+                //invalid issuer
+                invalidItems.add(new InvalidItemDTO(SAMLQueryRequestConstants.ValidationType.VAL_ISSUER,
+                        SAMLQueryRequestConstants.ValidationMessage.VAL_ISSUER_ERROR));
+                log.error(SAMLQueryRequestConstants.ValidationMessage.VAL_ISSUER_ERROR);
+                return false;
+            }
+            if (!isSignatureValidated) {
+                //invalid signature
+                invalidItems.add(new InvalidItemDTO(SAMLQueryRequestConstants.ValidationType.VAL_SIGNATURE,
+                        SAMLQueryRequestConstants.ValidationMessage.VAL_SIGNATURE_ERROR));
+                log.error(SAMLQueryRequestConstants.ValidationMessage.VAL_SIGNATURE_ERROR);
+            }
+        } catch (IdentityException ex) {
+            log.error(SAMLQueryRequestConstants.ServiceMessages.SERVER_ERROR_PROCESSING_ISSUER_SIG_VERSION);
+            return false;
+        }
+        return isSignatureValidated;
+    }
+
+    /**
+     * This method is used to validate signature
+     *
+     * @param request any type of assertion request message
+     * @return Boolean true, if signature is validated
+     */
+    protected boolean validateSignature(RequestAbstractType request) {
+        String alias;
+        boolean isValidSig;
+        String domainName;
+        alias = ssoIdpConfig.getCertAlias();
+        domainName = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+
+            isValidSig = OpenSAML3Util.validateXMLSignature(request,
+                    alias, domainName);
+
+            if (isValidSig) {
+                log.info("Signature successfully validated");
+                return true;
+
+            } else {
+                log.info("In valid Signature");
+                return false;
+            }
+
+        } catch (IdentityException ex) {
+            log.error(SAMLQueryRequestConstants.ServiceMessages.SIGNATURE_VALIDATION_FAILED);
```

Comment:
Better to log the exception along with the error message since you are not throwing it to an upper layer.